app/scripts/lib/backend-metametrics.js

```diff
@@ -0,0 +1,41 @@
+const {
+  getCurrentNetworkId,
+  getSelectedAsset,
+  getAccountType,
+  getNumberOfAccounts,
+  getNumberOfTokens,
+} = require('../../../ui/app/selectors/selectors')
+const {
+  sendMetaMetricsEvent,
+} = require('../../../ui/app/helpers/utils/metametrics.util')
+
+const inDevelopment = process.env.NODE_ENV === 'development'
+
+const METAMETRICS_TRACKING_URL = inDevelopment
+  ? 'http://www.metamask.io/metametrics'
+  : 'http://www.metamask.io/metametrics-prod'
+
+async function backEndMetaMetricsEvent (getState, eventData) {
+  const metamask = await getState()
```

Comment:
i personally find it easier to pass state object around instead of `getState`, and I would create a new selector called `getMetaMetricState` to get line 23 to 29, and passed it in to `backEndMetaMetricsEvent`. Based on code here I don't think it would mess with asynchronicity. What do you think @danjm ?

Reply:
Thanks @chikeichan. Changes made in 6b3c8c7cf